apt_offline_core/AptOfflineCoreLib.py

```diff
@@ -945,11 +946,16 @@ def fetcher( args ):
 
                 try:
                         for authstr in Str_HttpBasicAuth:
+                                # Add https schema if not provided any
+                                if not re.match('^[a-z]*:?//',authstr):
+                                        authstr = 'https://' + authstr
                                 # Add the username and password.
                                 parsedUrl = urllib.parse.urlparse(authstr)
                                 username = parsedUrl.username
-                                password = parsedUrl.password
+                                password = urllib.parse.unquote(parsedUrl.password)
                                 top_level_url = parsedUrl.hostname
+                                if parsedUrl.port:
+                                        top_level_url += ":" + str(parsedUrl.port)
```

Comment:
Is the scheme not needed when passing to password_mgr ?

Reply:
yup, it is not needed
However, the urlparser needs it... so i have to add it if it don't have (the re expression)

Reply:
The main issue with the urlparser is that if there is no schema, it can't find correctly username/password.

Reply:
> yup, it is not needed
> However, the urlparser needs it... so i have to add it if the url don't have (the re expression is for that)

That is why I added the helper string mentioning it clearly that the username/password needs to be fed in an expected manner, which is: **https://username:password@hostname.com:443/**

We need to make some syntactical assumption about what we accept as input from user. This format is our base assumption.